src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt

```diff
@@ -206,7 +206,9 @@ data class ClusterMetricsInput(
          * it auto encodes path params in the url.
          */
         return try {
-            val uriBuilder = URIBuilder("$SUPPORTED_SCHEME://$SUPPORTED_HOST:$SUPPORTED_PORT$path$pathParams")
```

Comment:
For context, this line was causing the issue. The comment suggests it was only added on the 3.0 branch because of an issue with the `httpcomponents 5.1.x` library.

It's not present on the [2.x](https://github.com/opensearch-project/common-utils/blob/2.x/src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt#L202) or [2.5](https://github.com/opensearch-project/common-utils/blob/2.5/src/main/kotlin/org/opensearch/commons/alerting/model/ClusterMetricsInput.kt#L202) branches; so this PR doesn't need to be backported.